modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/fields/modal_content/AddFieldsModalContent.tsx

```diff
@@ -101,24 +113,52 @@ function applyFilter({
 }: {fields?: Array<IFieldTreeItem>; query?: string} = {}) {
 	if (!query || !fields) {
 		return {
+			counter: 0,
 			filteredItems: fields ?? [],
 			filteredKeys: [],
 		};
 	}
 
+	let counter = 0;
 	const filteredKeys: Array<React.Key> = [];
-	const filteredItems = filterFields(fields, query, ({id}: IField) => {
-		if (id) {
-			filteredKeys.push(id);
-		}
+	const filteredItems = filterFields({
+		fields,
+		onFilter: ({id}: IField) => {
+			if (id) {
+				filteredKeys.push(id);
+			}
+		},
+		onMatch: () => counter++,
+		query,
 	});
 
 	return {
+		counter,
 		filteredItems,
 		filteredKeys,
 	};
 }
 
+const Highlight = ({query, text}: {query?: string; text?: string}) => {
+	if (!query || !text) {
+		return <>{text ?? ''}</>;
+	}
+
+	const indexMatch = text.search(RegExp(query, 'i'));
+
+	return indexMatch > -1 ? (
+		<>
+			{text.substring(0, indexMatch)}
+			<mark className="bg-transparent border-0 font-weight-bold p-0 shadow-none">
+				{text.substring(indexMatch, indexMatch + query.length)}
+			</mark>
```

Comment:
I'm not sure about `<mark>`. It really sticks out a lot, compared to other search solutions in DMS. I see your point about semantic correctness. I checked Lexicon, it has [this to say on search field](https://liferay.design/lexicon/core-components/forms/text-input-variations/#search-field):
> Characters in each entry must be semibold and marked in regular as the user types

So, semibold normally and normal text for hits? :sweat_smile:  

I'd put this alongside debouncing, as a potential improvement that we should apply consistently. I'd keep it normal bold for now.

Sidenote - I don't mind us being inconsistent with fuzzy, because there are many strong reasons against it, and it's not so noticeable.

Reply:
I don't completely understand you.

In the lexicon link, we are only talking about visuals :D 

The visual part of the `<mark>` is the same as using a `<strong>`.

how the `<marks>` looks after PR:

![image](https://github.com/liferay-frontend/liferay-portal/assets/67901/e119917d-ca99-4eff-aa90-99a5e74c7691)

I implemented the Highlight component to be as simple as possible to reuse it on the other DSM search/filter places. 

We can take advantage of it to spread the `mark` tag along the DSM if we decide to use the simplest higthligt.

The only concern to me is the abuse of CSS utility classes, but we can fit it in the future.